pkg/test/runner/config.go

```diff
@@ -0,0 +1,99 @@
+package runner
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"gopkg.in/yaml.v3"
+)
+
+// TestCaseConfig contains the config information for the test case
+type TestCaseConfig struct {
+	ExitCode int  `json:"exitCode,omitempty" yaml:"exitCode,omitempty"`
+	Network  bool `json:"network,omitempty" yaml:"network,omitempty"`
+	RunTimes int  `json:"runTimes,omitempty" yaml:"runTimes,omitempty"`
+}
+
+func newTestCaseConfig(path string) (TestCaseConfig, error) {
+	configPath := filepath.Join(path, expectedDir, expectedConfigFile)
+	b, err := ioutil.ReadFile(configPath)
+	if os.IsNotExist(err) {
+		// return default config
+		return TestCaseConfig{
+			ExitCode: 0,
+			Network:  false,
+			RunTimes: 2,
```

Comment:
We default RunTimes to 2 for functions catalog to ensure idempotency.
Does 2 still make sense for pipeline use case?

Reply:
Talked to Sunil and this is not a requirement for pipeline. As a result I've changed the default to 1, thanks for pointing this out. But that means on your functions-catalog tests you will have to specify your tests to run twice when you need to check idempotency